### PR TITLE
 Report on code coverage

### PR DESCRIPTION
**Pull request by esoergel**
## Summary

Once the build passes, codecov should add a check to the checks page.  I'm leaving the comment disabled, as it's kinda spammy.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: docker/run.sh [hunk 1]
```diff
@@ -159,8 +159,15 @@ function _run_tests {
 
     function _test_python {
         ./manage.py create_kafka_topics
-        logmsg INFO "./manage.py test ${py_test_args[*]}"
-        ./manage.py test "${py_test_args[@]}"
+        if [ "$TRAVIS_EVENT_TYPE" ]; then
```
Comment (millerdev) [inline docker/run.sh:162]: This doesn't look like valid bash syntax to me. Did you intend to do
```suggestion
        if [ -n "$TRAVIS_EVENT_TYPE" ]; then
```

**Review by millerdev — APPROVED** [review]

**Comment by millerdev** [comment]
When I try to click through to the codecov report I see

image

Do I need to do something special to view the report?

File: docker/run.sh (hunk 1 above)
Comment (esoergel) [reply docker/run.sh]: I've got no problems sticking in an `-n` for clarity

**Comment by esoergel** [comment]
> When I try to click through to the codecov report I see...

I'm not sure what the rate limit error is about - you might need to sign in to avoid rate limiting.  Are you able to view [this link](https://app.codecov.io/gh/dimagi/commcare-hq/)? Codecov has seemed to have problems finding the PR - I believe the reason I've seen fluctuation in overall coverage is because codecov doesn't correctly identify all four build artifacts as coming from the same build.  After some digging, I think it's because the info is submitted from within a docker container, so it doesn't have access to all the [travis environment variables](https://docs.travis-ci.com/user/environment-variables#default-environment-variables).  I added a commit to pass through the relevant env vars, which should hopefully resolve that issue.

**Comment by millerdev** [comment]
Weird. Your link loads correctly, but when I try to click through to the report for this PR now I get a new error:

> Oops, something went wrong
> 
> Changes found in between 6aba491...795c4cf (pseudo...base) which prevent comparing this pull request. Check on Codecov's status or see our docs for common support.
> 
> Error 400

**Comment by esoergel** [comment]
@millerdev I think I've at last provided enough context for codecov to reliably link together all four job artifacts.

Regarding the error, I believe that's because codecov doesn't have an adequate base commit for comparison.  I'm hoping once this is merged and we start sending a lot more commits to codecov, their algorithm for finding a pseudo-base will be effective.

**Review by millerdev — APPROVED** [review]

**Pull request merged**